src/backend/tcop/zenith_wal_redo.c

```diff
@@ -313,24 +318,23 @@ WalRedoMain(int argc, char *argv[],
 	/*
 	 * Main processing loop
 	 */
+	MemoryContextSwitchTo(MessageContext);
+	initStringInfo(&input_message);
+
 	for (;;)
 	{
 		/*
 		 * Release storage left over from prior query cycle, and create a new
 		 * query input buffer in the cleared MessageContext.
```

Comment:
"cleared MessageContext" isn't accurate anymore, as we don't reset in the loop anymore.

Reply:
Will fix this comment